webtransport: disable support for signed/static TLS certificates

MarcoPolo opened the pull request:
This needs more thought before we enable support.

marten-seemann reviewed (approved):
Should we `t.Skip` the tests instead of deleting them? We do want this feature at some point in the future.

It would probably also makes sense to create an issue in this repo, and include a link in the error message and in the skip message.

marten-seemann commented:
Yes.

MarcoPolo commented:
If we use t.Skip we have to make changes since we removed the `WithTLSConfig` option. So I removed them. I think it's pretty easy to revive dead code from the git history, so I would prefer to keep it deleted and revive it when needed.
